org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -0,0 +1,265 @@
+import org.eclipse.lsp4xml.utils.FilesUtils;
+import org.eclipse.lsp4xml.utils.StringUtils;
+
+/**
+ * FilePathCompletionParticipant
+ */
+public class FilePathCompletionParticipant extends CompletionParticipantAdapter {
+
+	public static final String FILE_SCHEME = "file";
+
+
+	@Override
+	public void onTagOpen(ICompletionRequest completionRequest, ICompletionResponse completionResponse) throws Exception {
+
+	}
+
+	@Override
+	public void onXMLContent(ICompletionRequest request, ICompletionResponse response) throws Exception {
+
+	}
+
+	@Override
+	public void onAttributeName(boolean generateValue, Range fullRange, ICompletionRequest request,
+			ICompletionResponse response, SharedSettings settings) throws Exception {
+
+	}
+
+	@Override
+	public void onAttributeValue(String valuePrefix, Range fullRange, boolean addQuotes, ICompletionRequest request,
+			ICompletionResponse response, SharedSettings settings) throws Exception {
+
+		DOMDocument xmlDocument = request.getXMLDocument();
+		String text = xmlDocument.getText();
+
+		// Get full attribute value range
+		// + 1 since it includes the quotations
+		int documentStartOffset = xmlDocument.offsetAt(fullRange.getStart()) + 1;
+		
+		String fullAttributeValue = valuePrefix;
+		if (isEmpty(fullAttributeValue)) {
+			return;
+		}
+
+		// Get value and range from fullAttributeValue
+		int completionOffset = request.getOffset(); // offset after the typed character
+		int parsedAttributeStartOffset = StringUtils.getOffsetAfterWhitespace(fullAttributeValue, completionOffset - documentStartOffset) + documentStartOffset; // first character of URI
+		String subAttributeValue = text.substring(parsedAttributeStartOffset, completionOffset);
+	
+		Position startValue = xmlDocument.positionAt(parsedAttributeStartOffset);
+		Position endValue = xmlDocument.positionAt(completionOffset);
+		fullRange = new Range(startValue, endValue);
+
+		// Try to get the URI string from the attribute value in case it has a file scheme
+		// header (eg: "file://")
+		String pathInAttribute = subAttributeValue;
+		boolean hasFileScheme = false;
+
+		hasFileScheme = subAttributeValue.startsWith(FilesUtils.FILE_SCHEME);
+		if (hasFileScheme) {
+			pathInAttribute = subAttributeValue.substring(FilesUtils.FILE_SCHEME.length());
+		}
+
+		String slash = getFilePathSlash(subAttributeValue);
+
+		if (hasFileScheme) {
+			if (!pathInAttribute.startsWith("/")) {
+				return; // use of 'file://' and the path was not absolute
+			}
+			if (isWindows && pathInAttribute.length() == 1) { // only '/', so list Windows Drives
+				
+				Range replaceRange = adjustReplaceRange(xmlDocument, fullRange, subAttributeValue, "/");
+
+				File[] drives = File.listRoots();
+				for (File drive : drives) {
+					createFilePathCompletionItem(Paths.get("/"), drive, replaceRange, response, "/");
+				}
+				return;
+			}
+		}
+		else {
+			if(isWindows && "/".equals(slash)) { // non-URI using forward slash in Windows
+				return;
```

Comment:
The '/' using must work with Windows OS too because we can use it in noSchemaLocation. In other words filr path completion must have the same behaviour for any OS.

Reply:
Forget my comments